src/document.h

```diff
@@ -1117,12 +1111,36 @@ struct Document {
                 if (sys->frame->borddd) sys->frame->borddd->ShowPopup();
                 break;
 
+            case A_REPLACEONCE:
+            case A_REPLACEONCEJ:
             case A_REPLACEALL: {
                 if (!sys->searchstring.Len()) return _(L"No search.");
-                rootgrid->AddUndo(this);  // expensive?
-                rootgrid->FindReplaceAll(sys->frame->replaces->GetValue());
-                rootgrid->ResetChildren();
-                Refresh();
+                if (k == A_REPLACEALL) {
+                    wxString filter = sys->frame->filter->GetValue();
+                    wxString replaces = sys->frame->replaces->GetValue();
+                    loopallcells(c) if(c->text.IsInSearch()) c->AddUndo(this);

```

Comment:
Hmm this is also potentially a lot.. we may need something better for this, but I guess that can be a future change..